### ISSUE TITLE
Split IService to simpler interfaces like IDataProxy

### ISSUE THREAD

**Issue by SlowLogicBoy** [ahanusa/Peasy.NET#7]
I have some `IDomainObject`'s that can only be selected (`GetAll`, `GetById` and, `GetByFilter`), however right now I only know if they can be inserted / deleted by calling that command and it throws `NotSupportedException`.

`IService` should be separated into smaller interfaces like `IDataProxy`.
Also `DataProxy` should have non generic `IDataProxy` implementation like `IDomainObject`, same with `IService`

**Comment by SlowLogicBoy** [comment]
I can't think of a name for common interface for ISupport* CRUD interfaces :/.

**Comment by ahanusa** [comment]
Sorry for taking so long to get back to you on your PRs.  I was on vacation and needed to disconnect for a while.

So I accidentally closed your other PR without commenting, so I'll comment on that here.  I decided to close the filter story because it's really not something that I feel belongs in the peasy framework. 

While many people could benefit from that interface, it's not something that everyone needs.  Adding that interface will cause breaking changes and many users have already implemented their own filtering logic by adding it to their own base classes that inherit from peasy base classes.

I don't want to force people to implement that interface when they most likely won't need it.  Hopefully you understand?

Which brings me to this story, and you're right.  IService could definitely be broken out into smaller interfaces.  Doing so would also not present breaking changes, which I like.  I never actually envisioned a service not being used for all CRUD operations, so you state a valid use case.

So basically, this would allow you to inject your service as a particular ISupportGetXZYCommand, and then never have to worry about whether or not your service implements anything else.

I don't particularly see the need for your service extensions class in the peasy framework, so maybe remove that from the PR and I can merge in the changes to IService after some test coverage has been added.

**Comment by ahanusa** [comment]
I forgot to comment on "DataProxy should have non generic IDataProxy implementation like IDomainObject".  Can you give me more details on that?  I'm not sure what you mean.

Also, thanks for all of your work on this, I appreciate it!

**Comment by SlowLogicBoy** [comment]
> I forgot to comment on "DataProxy should have non generic IDataProxy implementation like IDomainObject". Can you give me more details on that? I'm not sure what you mean.

With simple IDataProxy interface I could get all DataProxies by using IDataProxy interface through DI,

Lets say we add this interface to all sub interfaces (GetAll, GetByID etc..) by resolving IDataProxy, I could get all classes that can handle data directly with database, and this way If I add ISupportFilter : IDataProxy
Then I can also get my classes that implement my ISupportFilter interface.

Also if you saw in my service PR, I also made an empty IService interface, which allows me to get all classes, that do any Service related task.

**Comment by ahanusa** [comment]
I see, you just want to use it as a marker interface.  I don't really think it serves a purpose in the peasy framework, so I would suggest you just define that interface in your own library and implement as you see fit.  The same can be applied to the IService marker interface.
